File: backend/app/models/enums.py
"""Enum types shared by ORM models and Pydantic schemas.

Values mirror the `## Enums` section of docs/API_CONTRACT.md exactly. Both the
SQLAlchemy models and the Pydantic schemas import these same enums so there is
a single source of truth for the wire format.
"""

import enum


class ProjectType(str, enum.Enum):
    MANGROVE_RESTORATION = "mangrove_restoration"
    FOREST_CONSERVATION = "forest_conservation"
    AGROFORESTRY = "agroforestry"
    WETLAND_RESTORATION = "wetland_restoration"
    GRASSLAND_RESTORATION = "grassland_restoration"


class ProjectStatus(str, enum.Enum):
    PLANNING = "planning"
    ACTIVE = "active"
    MONITORING = "monitoring"
    COMPLETED = "completed"


class EcosystemType(str, enum.Enum):
    MANGROVE = "mangrove"
    FOREST = "forest"
    WETLAND = "wetland"
    GRASSLAND = "grassland"
    AGROFORESTRY = "agroforestry"


class MonitoringStatus(str, enum.Enum):
    ACTIVE = "active"
    PAUSED = "paused"
    NEEDS_REVIEW = "needs_review"
    COMPLETED = "completed"


class ObservationCategory(str, enum.Enum):
    BIRD = "Bird"
    MAMMAL = "Mammal"
    REPTILE = "Reptile"
    AMPHIBIAN = "Amphibian"
    PLANT = "Plant"
    INSECT = "Insect"


class DisturbanceRisk(str, enum.Enum):
    LOW = "low"
    MODERATE = "moderate"
    HIGH = "high"


class AnalyticsRange(str, enum.Enum):
    THREE_MONTHS = "3m"
    SIX_MONTHS = "6m"
    TWELVE_MONTHS = "12m"
    ALL = "all"


# --- New enums for FEATURE_CONTRACT_V2 (Nature Health Score & Alerts, Field
# Observations, Conservation Actions, Audit Log). Values mirror that contract
# exactly, mirroring the pattern used for the v1 enums above. ---


class ScoreBand(str, enum.Enum):
    HEALTHY = "healthy"
    WATCH = "watch"
    AT_RISK = "at_risk"
    CRITICAL = "critical"


class AlertSeverity(str, enum.Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    CRITICAL = "critical"


class AlertStatus(str, enum.Enum):
    OPEN = "open"
    ACKNOWLEDGED = "acknowledged"
    RESOLVED = "resolved"


class ObservationType(str, enum.Enum):
    SPECIES = "Species"
    HABITAT = "Habitat"
    THREAT = "Threat"
    RESTORATION_ACTIVITY = "Restoration Activity"
    OTHER = "Other"


class ActionCategory(str, enum.Enum):
    FIELD_SURVEY = "Field Survey"
    HABITAT_RESTORATION = "Habitat Restoration"
    COMMUNITY_ENGAGEMENT = "Community Engagement"
    MONITORING = "Monitoring"
    RISK_INVESTIGATION = "Risk Investigation"


class ActionPriority(str, enum.Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    CRITICAL = "critical"


class ActionStatus(str, enum.Enum):
    PLANNED = "planned"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    CANCELLED = "cancelled"

File: frontend/src/types/enums.ts
// String literal union types mirroring the enums section of docs/API_CONTRACT.md exactly.
// Do not change these values without updating the backend contract first.

export type ProjectType =
  | 'mangrove_restoration'
  | 'forest_conservation'
  | 'agroforestry'
  | 'wetland_restoration'
  | 'grassland_restoration'

export type ProjectStatus = 'planning' | 'active' | 'monitoring' | 'completed'

export type EcosystemType = 'mangrove' | 'forest' | 'wetland' | 'grassland' | 'agroforestry'

export type MonitoringStatus = 'active' | 'paused' | 'needs_review' | 'completed'

export type ObservationCategory = 'Bird' | 'Mammal' | 'Reptile' | 'Amphibian' | 'Plant' | 'Insect'

export type AnalyticsRange = '3m' | '6m' | '12m' | 'all'

export type DisturbanceRisk = 'low' | 'moderate' | 'high'

export const PROJECT_TYPES: ProjectType[] = [
  'mangrove_restoration',
  'forest_conservation',
  'agroforestry',
  'wetland_restoration',
  'grassland_restoration',
]

export const PROJECT_STATUSES: ProjectStatus[] = ['planning', 'active', 'monitoring', 'completed']

export const ECOSYSTEM_TYPES: EcosystemType[] = [
  'mangrove',
  'forest',
  'wetland',
  'grassland',
  'agroforestry',
]

export const MONITORING_STATUSES: MonitoringStatus[] = [
  'active',
  'paused',
  'needs_review',
  'completed',
]

export const OBSERVATION_CATEGORIES: ObservationCategory[] = [
  'Bird',
  'Mammal',
  'Reptile',
  'Amphibian',
  'Plant',
  'Insect',
]

export const ANALYTICS_RANGES: AnalyticsRange[] = ['3m', '6m', '12m', 'all']

export const DISTURBANCE_RISKS: DisturbanceRisk[] = ['low', 'moderate', 'high']
